engine/src/main/java/org/terasology/engine/rendering/primitives/BlockMeshGeneratorSingleShape.java

```diff
@@ -24,7 +24,10 @@ public BlockMeshGeneratorSingleShape(Block block) {
 
     @Override
     public void generateChunkMesh(ChunkView view, ChunkMesh chunkMesh, int x, int y, int z) {
-        final Block selfBlock = view.getBlock(x, y, z);
+        final BlockAppearance blockAppearance = block.getPrimaryAppearance();
+        if (!blockAppearance.hasAppearance()) {
+            return;
+        }
```

Comment:
we were just evaluating all air blocks so we check the ordinals of all the empty blocks :/

Reply:
this is useful information we should add as docstring comment.